### DISCUSSION TITLE
How to run tests on Sepolia testnet in Foundry?

### DISCUSSION THREAD

**Discussion by JanSzczepan** [Cyfrin/foundry-full-course-cu#774] [Q&A]
I know that it is possible to run tests on forked testnet in Foundry
```
forge test --fork-url SEPOLIA_RPC_URL
```
or
```
forge test --rpc-url SEPOLIA_RPC_URL #same result with --fork-url
```
But what if I want to run my tests on real sepolia testnet and actually pay sepolia eth for this tests? Is there a way in Foundry to do this (I know that in hardhat it is easily possible).

I have even this modifier in my tests that should only run on the real testnets:
```solidity
modifier onlyOnDeployedContracts() {
      if (block.chainid == 31337) {
         return;
      }
      try vm.activeFork() returns (uint256) {
         return;
      } catch {
         _;
      }
   }
```
vm.activeFork should revert if there is no forking.

**Comment by olahamid** [comment]
can you share the error you are getting? i know rpc-url is for reallive testnet, while fork-url is for dummy(mock) test. so i guess rpc-url should work.

**Comment by Niferu** [comment]
@olahamid 
He doesnt encounter any errors..

@JanSzczepan 
I believe there is no way to test it as on hardhat. At least documentation doesnt say anything about direct testing, only local or fork, but I can be wrong.
